app/containers/MessageActions/index.js

```diff
@@ -179,7 +179,13 @@ const MessageActions = React.memo(forwardRef(({
 
 	const handlePin = async(message) => {
 		try {
-			await RocketChat.togglePinMessage(message.id, message.pinned);
+			const db = database.active;
+			await db.action(async() => {
+				await message.update((m) => {
+					m.pinned = !m.pinned;
+				});
+			});
+			await RocketChat.togglePinMessage(message.id, !message.pinned);
```

Comment:
We should go back to the original state on database when an error happens

Reply:
How can I do that? Can you explain that please @djorkaeffalexandre ?

Reply:
@djorkaeffalexandre Can you help me a bit and explain, How should we do that?